src/components/App/App.js

```diff
@@ -1,89 +1,192 @@
 import './App.css';
-import React, { useState } from 'react';
+import React, { useEffect, useState } from 'react';
 import { Routes, Route } from 'react-router-dom';
 
+import * as auth from '../../utils/MainApi.js';
+import { CurrentUserContext } from '../../contexts/CurrentUserContext';
 import Main from '../Main/Main';
 import Header from '../Header/Header';
 import Footer from '../Footer/Footer';
 import SavedNews from '../SavedNews/SavedNews';
 import SignInPopup from '../SignInPopup/SignInPopup';
 import PopupWithText from '../PopupWithText/PopupWithText';
 import SignUpPopup from '../SignUpPopup/SignUpPopup';
+import ProtectedRoute from '../ProtectedRoute/ProtectedRoute';
+import { useFormWithValidation } from '../../hooks/useFormWithValidation/index.js';
+
+import { newsApi } from '../../utils/NewsApi';
 
 function App() {
   const [isSignInPopupOpen, setIsSignInPopupOpen] = useState(false);
   const [isSignUpPopupOpen, setIsSignUpPopupOpen] = useState(false);
   const [isSuccessPopupOpen, setIsSuccessPopupOpen] = useState(false);
+  const [currentUser, setCurrentUser] = useState({});
+  const [loggedIn, setLoggedIn] = useState(false);
+  const [loggedInData, setLoggedInData] = useState(null);
+
+  const signInValidation = useFormWithValidation({
+    password: '',
+    email: '',
+  });
+  const signUpValidation = useFormWithValidation({
+    password: '',
+    email: '',
+    username: '',
+  });
+
+  const getNews = async (q) => {
+    return await newsApi.getNews(q);
+  };
 
-  const handleSwitchPopupClick = () => {
-    setIsSignUpPopupOpen(!isSignUpPopupOpen);
-    setIsSignInPopupOpen(!isSignInPopupOpen)
+  const getSavedNews = async () => {
+    const jwt = localStorage.getItem('jwt');
+    if (jwt) {
+      return await auth.getArticles(); 
+    }
   };
-  
-  const handleonSignInClick = () => {
+
+  useEffect(() => {
+    const jwt = localStorage.getItem('jwt');
+    if (jwt) {
+      auth
+        .checkToken(jwt)
+        .then((res) => {
+          if (res) {
+            const loggedInData = res;
+            setLoggedIn(true);
+            setLoggedInData(loggedInData.data);
+            setCurrentUser({ jwt: res, isLoggedIn: true });
+          }
+        })
+        .then(() => {
+          setLoggedIn(true);
+        });
```

Comment:
You need to `catch` possible errors at the very end of any server request.

**Please, correct it everywhere**